lib/rules/no-js-extension.js

```diff
@@ -20,11 +20,35 @@ module.exports = function (context) {
 
     var MESSAGE = "Don't use .js extension in dependency path.";
 
+    var pluginsToCheck = context.options[0] || [];
+
     //--------------------------------------------------------------------------
     // Helpers
     //--------------------------------------------------------------------------
 
     /**
+     * Determines if provided `node` should be checked for .js extension or not.
+     *
+     * @private
+     * @param {ASTNode} node - Literal node to test
+     * @returns {Boolean} true if literal node should be checked for .js extension
+     */
+    function shouldBeChecked(node) {
+        var result = true,
+            pluginPrefixRegExp = /^(\w+)\!/i,
```

Comment:
Also I'm wondering if you could build the regex once dynamically based on `pluginsToCheck`? So for a plugin list like `['foo', 'bar']` you might end up with `/^(foo|bar)\!/i`. This might allow you to get rid of the loop below...